src/approaches/grammar_search_invention_approach.py

```diff
@@ -469,6 +469,26 @@ def enumerate(self) -> Iterator[Tuple[Predicate, float]]:
 class _PrunedGrammar(_DataBasedPredicateGrammar):
     """A grammar that prunes redundant predicates."""
     base_grammar: _PredicateGrammar
+    _state_sequences: List[List[State]] = field(init=False,
+                                                default_factory=list)
+
+    def __post_init__(self) -> None:
+        if CFG.segmenter == "option_changes":
+            # If the segmenter is based on changes in options, then
+            # because it doesn't depend on atoms, we can be very
+            # efficient during pruning by pre-computing the segments.
+            # Then, we only need to care about the initial and final
+            # states in each segment, which we store into
+            # self._state_sequence.
+            atom_dataset = utils.create_ground_atom_dataset(
+                self.dataset.trajectories, set())
+            for traj in atom_dataset:
+                if not traj[0].is_demo:
```

Comment:
the pruner here should probably be agnostic to demo vs not demo, so remove this line

Reply:
we can't bc segmentation would crash on any traj that isn't a demo, since it doesn't have an option (this came up during unit testing...)

Reply:
got it, but this still doesn't feel like the right check. I think that crash is actually good/informative. can we just adjust the unit test?

Reply:
kk